bundles/org.openhab.core.thing.xml/src/main/java/org/openhab/core/thing/xml/internal/ChannelTypeConverter.java

```diff
@@ -70,29 +74,30 @@ private boolean readBoolean(Map<String, String> attributes, String attributeName
     }
 
     private String readItemType(NodeIterator nodeIterator) throws ConversionException {
-        return (String) nodeIterator.nextValue("item-type", false);
+        return requireNonEmpty((String) nodeIterator.nextValue("item-type", false),
+                "ChannelType 'itemType' must not be null or empty.");
```

Comment:
@wborn I think this change causes #2831. For trigger-channels an empty `itemType` is ok.